### ISSUE TITLE
Column-major problems

### ISSUE THREAD

**Issue by Goobley** [MAVENSDC/cdflib#59]
I am in the process of converting my RadynPy package from `spacepy.pycdf` to `cdflib`. SpacePy's library relies on the compiled CDF library, which is a dependency I would rather not have, and SpacePy also seems to depend on an old version of NumPy, causing problems with installation.
I compared the results from reading the CDF files produced by a simulation (RADYN (FORTRAN77), sample files [here](https://star.pst.qub.ac.uk/wiki/doku.php/public/solarmodels/start)) and found that whilst the dimensions were correct (albeit not for 2D arrays of strings), the data appeared to have been assumed to be in row-major format when converted to numpy arrays, compared to the expected form and what SpacePy produces.
As an example, for the `g` variable in any of those files. It should appear as
```
array([[ 2.,  2.,  1.],
       [ 8.,  4.,  3.],
       [18.,  6.,  1.],
       [32.,  2.,  9.],
       [50.,  4.,  3.],
       [ 1.,  1.,  2.],
       [ 0.,  0.,  2.],
       [ 0.,  0.,  6.],
       [ 0.,  0.,  2.]])
```

but is instead read as
```
array([[ 2.,  8., 18.],
       [32., 50.,  1.],
       [ 0.,  0.,  0.],
       [ 2.,  4.,  6.],
       [ 2.,  4.,  1.],
       [ 0.,  0.,  0.],
       [ 1.,  3.,  1.],
       [ 9.,  3.,  2.],
       [ 2.,  6.,  2.]])
```

I presume this is not the expected behaviour, since the CDF library handles these files correctly. I could not see any differences in the way the program handles arrays of different majority, so I made some basic fixes in 520d7b6. I am happy to submit this as a PR and find a way to make these changes work for mainline `cdflib` if you would be interested in working through this, as I would rather not maintain my own dependency, although we would have to discuss if it is acceptable to your use cases to convert arrays of strings from flat python lists to numpy arrays.
Many thanks!

**Comment by bryan-harter** [comment]
Ah thanks for pointing that out, MAVEN's cdfs are apparently always row-major so I haven't noticed that code literally didn't exist to handle it!

I just pushed something to master that I believe is slightly simpler, it just flattens the array and rebuilds it as column-major if the CDF is column major.  Let me know if you see an issue with that though.  

I'm not entirely sure what to do with the strings yet though, but I suppose there is no reason that those numpy arrays wouldn't work

**Comment by Goobley** [comment]
Thanks very much for looking at this. I can confirm that the numeric column-major arrays are now being read correctly.
Depending on your plan to handle the strings, I can put them into an array and reshape it myself in `radynpy` if you would like to retain the list behaviour (as they are being read into a flat list in the right order now). I understand that `numpy` arrays of strings can be quite problematic if somebody wants to increase the length of an element unless the dtype is set to object (which could be a slow way of handling this if anyone else needs 2D+ arrays of strings. (In our case we just read them from the file, so the fixed allocation length is fine).

**Comment by bryan-harter** [comment]
Actually my code did not work when reading in multiple records, as I forgot that the record number was not considered a dimension in CDF files.  So you had the right idea by not transposing the first dimension in your code.  

I reverted my change and added all of yours in, including the handling of the strings for now.  I also pushed a new version up to pypi.  

Thanks a lot for finding and solving this issue!
